### PR TITLE
Added a method to know whether a given method is part of a CtType

### PR DESCRIPTION
**Pull request by arnobl**
Hey all,
As suggested on the mailing list, here is a PR to add a method in CtType to know whether a given type has a method (inherited or not, abstract or implemented). Look at the Javadoc for more details. Let me know if you want me to change the behaviour of the added method (replacing PR 870).

### CODE REVIEW

**Comment by monperrus** [comment]
This method is quite handy to know whether one can call a method on typed variable.

I wonder if using a signature-based test would be better than an object-based test.

Ie replacing `method.getParent() == this` by
```
final String sig = method.getSignature();
for(CtMethod<?> m : getMethods()) {
	if(m.getSignature().equals(sig)) {
		return true;
	}
}
```
Obviously, this behavior supersedes the current one.

What do you think?

**Comment by arnobl** [comment]
Yes, the object-based test is not mandatory but just an optimisation. I can remove it. Tell me.

**Comment by monperrus** [comment]
let's go for a signature-based test then

File: src/main/java/spoon/reflect/declaration/CtType.java [hunk 1]
```diff
@@ -216,6 +216,14 @@
 	List<CtMethod<?>> getMethodsByName(String name);
 
 	/**
+	 * Searches in the type for the given method.
+	 * Super classes and implemented interfaces are considered.
```
Comment (monperrus) [inline src/main/java/spoon/reflect/declaration/CtType.java]: The matching criterion is that the signatures are identical.

**Comment by monperrus** [comment]
thanks a lot for this contribution

**Pull request merged**